### ISSUE TITLE
Accessibility issues with ion-fab button 

### ISSUE THREAD

**Issue by mastergogo** [ionic-team/ionic-framework#22296]
Below are a few observations for improvements:

When checked with Iphone's screen reader (Voice over), it has been observed that fab buttons are always announced as "close (icon name) button" when the user double taps on the buttons, more interactive options appears such as social media icons when navigating on the social media icons, Voice over announces as "close (social media icon name)", issues here is that voice-over announces close every time user interacts with it which is an incorrect announcement until the close button appears it should not announce close, it should only announce the name and the role of a button that is currently in focus. 

below are steps to reproduce:

1) please open the [link](https://ionic-docs-demo.herokuapp.com/component/fab) in Apple device such as iPhone and turn on voice over
2) once the voice-over is on, please navigate to the fab-button and observe the announcement 
3) it will be announced as a "close (icon-name) button", however, the close button is behind the fab button which is not activated yet but the voiceover still announces it.
4) when double tapped on the fab button, voice-over announces the close button as close, however, it announces all the social media icons as close, which is again incorrect.  

Let me know if more info is needed to reproduce.

Thanks,
Amit

**Comment by RenAlbiorix** [comment]
Is there any update when this issue can be fixed? All ion-fab button reads as close because inside shadow-root `<ion-icon part="close-icon" ...` is getting added regardless of icon used in ion-fab button. Below is screenshot from ionic site.
![image](https://user-images.githubusercontent.com/70148715/168017285-bca55b04-0379-43f6-bde1-66caa4c38f53.png)
